Index: application/controllers/cargo.php
===================================================================
--- application/controllers/cargo.php	(revision 5331)
+++ application/controllers/cargo.php	(working copy)
@@ -4740,29 +4740,25 @@
         $this->is_logged_in();
         $data = array();
         list($stations, $routes) = get_routes_and_stations();
-    $data['stations'] = $stations;
+        $data['stations'] = $stations;
 
-    $branches = $this->api_model->getOrganization();
-    $branches = json_decode($branches);
-    $branches = $branches->data;
+        $branches = $this->api_model->getOrganization();
+        $branches = json_decode($branches);
+        $branches = $branches->data;
 
-    $organization = array();
-    foreach ($branches as $branch) {
-        $organization[$branch->code] = $branch->name;
-    }
-    asort($organization);
-    $data['branches'] = $organization;
-
-    $data['cargo_status'] = config_item('cargo_status');
-    $data['cargo_payment_status'] = config_item('cargo_payment_status');
-
-        
+        $organization = array();
+        foreach ($branches as $branch) {
+            $organization[$branch->code] = $branch->name;
+        }
+        asort($organization);
+        $data['branches'] = $organization;
+        $data['cargo_status'] = config_item('cargo_status');
+        $data['cargo_payment_status'] = config_item('cargo_payment_status');
         $this->load->view('site/cargo/delivery-ledger-report', $data);
-}
+    }   
 
 
-public function delivery_ledger_report_list()
-    {
+    public function delivery_ledger_report_list(){
         $this->is_logged_in();
         $user_code = $this->input->post('userCode');
         if (empty($user_code)) {
@@ -4772,8 +4768,6 @@
         $from_date = date('Y-m-d', strtotime($fromDate));
         $to_date = date('Y-m-d', strtotime($toDate));
 
-
-
         $findOgplCode = $this->input->post('code');
         $ogplcargo = "";
         $odometer = "";
@@ -4865,8 +4859,6 @@
         $data['toDate'] = date('d-m-Y', strtotime($toDate));
         $data['user_profile'] = $usr_pro->data;
         $data['organizationDetails']  = $organizationDetails->data;
-
-
         foreach ($singleOgpl as $row) {
             if ($row->cargo_status_code == 'CA' || $row->cargo_status_code == 'CR') {
                 continue;
@@ -4885,7 +4877,7 @@
             $delivery_details[$row->delivery_type_code]['count'] += 1;
             $delivery_details[$row->delivery_type_code]['amount'] += $row->total_transaction_amount;
         
-    }
+        }
     
         $data['singleOgpl'] = $singleOgpl;
         $data['payment_details'] = $payment_details;
@@ -4911,7 +4903,6 @@
             echo $content;
             die;
         } else if ($this->input->post('print') == 1) {
-
             $this->load->exclude_template();
             if ($this->session->userdata('namespace_id') == "seenutrans" || $this->session->userdata('namespace_id') == "seenutransport") {
                 $this->load->view('site/cargo/delivery-report-print-seenutransport', $data);
@@ -4923,9 +4914,6 @@
         }
     }
 
-
-
-
     public function delivery_report_list_tally() {
         $this->is_logged_in();
         $this->load->library('excel');
@@ -8077,7 +8065,7 @@
         $branches = $this->api_model->getCargoOrganization();
         $branches = json_decode($branches);
         $data['branches'] = $branches->data;
-
+        
         $ownershipTypes = array();
         foreach ($vehicles->data as $vehicle) {
             $ownershipTypes[$vehicle->ownershipType->code] = $vehicle->ownershipType->name;
@@ -8605,14 +8593,12 @@
                 return;
             }
         }
+
+        $fromOrganization = $this->input->post('fromOrganization');
         $toOrganization = $this->input->post('toOrganization');
-
-        if(isset($toOrganization)){
-            $toOrganization= $toOrganization['code'];
-        }else{
-            $toOrganization=array('code' => $this->session->userdata('org_code'));
-        }
-
+        $fromOrganizationCode = isset($fromOrganization['code']) ? $fromOrganization['code'] : 'NA';
+        $toOrganizationCode = isset($toOrganization['code']) ? $toOrganization['code'] : 'NA';
+            
         $ogpl = $this->api_model->cargoOGPLupdate(array(
             'activeFlag' => $this->input->post('activeFlag'),
             'code' => $this->input->post('code') ?: '',
@@ -8619,8 +8605,8 @@
             'name' => $this->input->post('name'),
             'fromStation' => $this->input->post('fromStation'),
             'toStation' => $this->input->post('toStation'),
-            'fromOrganization' => array('code' => $this->session->userdata('org_code')),
-            'toOrganization' => array('code' => $toOrganization),
+            'fromOrganization' => array('code' => $fromOrganizationCode?$fromOrganizationCode:"NA"),
+            'toOrganization' => array('code' => $toOrganizationCode?$toOrganizationCode:"NA"),
             'viaStations' => $this->input->post('viaStations') ?: [],
             'viaZones' => $this->input->post('viaZones') ?: [],
             'hubViaStations' => $this->input->post('linkLoadStations') ?: [],
@@ -9430,11 +9416,21 @@
 
     public function ogpl_chart_details() {
         $this->is_logged_in();
-
+        $data['login_user'] = $this->session->userdata('user_id');
         $transitCode = $this->input->get('transitCode');
         $transitType = $this->input->get('transitType');
         $this->session->set_userdata('ogpl_last_active', $transitCode);
-
+        $data['login_branch'] = $this->session->userdata('org_code');
+        $odoDetails = $this->api_model->getTransitOdometer(array(
+            'transitCode' => $transitCode
+        ));
+        $odoDetails = json_decode($odoDetails);
+        foreach ($odoDetails->data as $row) {
+            if ($row->organization->code == $data['login_branch']) {
+                $data['odoDetails'] = $row;
+                break;
+            }
+        }
         $ogpl = $this->api_model->cargoOGPLDetails(array(
             'transitCode' => $transitCode
         ));
@@ -9441,6 +9437,9 @@
         $ogpl = json_decode($ogpl);
         $data['ogpl'] = $ogpl->data;
         $organizationCode = $this->session->userdata('org_code'); 
+        $vehicles = $this->api_model->getVehicle();
+        $vehicles = json_decode($vehicles);
+        $data['vehicles']=$vehicles->data;
 
         $response = $this->helper_model->view_chart($transitType,array(
             'queryCode' => 'RQJ45G4144',
@@ -9456,6 +9455,9 @@
         $data['result'] = $response['list'];
         $data['routes'] = $response['routes'];
         $data['summary'] = $response['summary'];
+        // echo "<pre>";
+        // print_r($response);
+        // exit;
         $data['rowCount'] = $response['rowCount'];
         
         if (_get_namespace_id() == 'newlinetrans') {
@@ -9522,7 +9524,7 @@
             $data['namespaceCode'] = $this->session->userdata('namespace_id');
             $this->load->view('site/cargo/ogpl-chart-cityparcel', $data);
         } else {
-        $this->load->view('site/cargo/ogpl-chart', $data);
+            $this->load->view('site/cargo/ogpl-chart', $data);
         }
     }
 
@@ -9960,7 +9962,9 @@
                 $odometer[$row->transitCargo->cargoActivityType->code][$row->fromStation->name] = $row;
             }
             $data['odoDetails'] = $odometer;
-            
+            // echo"<pre>";
+            // print_r($odometer);
+            // exit; 
             $this->load->view('site/cargo/out-for-delivery-hire-challan-print-seenutransport', $data);
         }
     }
@@ -13680,7 +13684,9 @@
         $data['cargo_status'] = config_item('cargo_status');
         $data['cargo_payment_status'] = config_item('cargo_payment_status');
         $data['delivery_type_codes'] = config_item('delivery_type_codes');
-
+        // echo "<pre>";
+        // print_r($data);
+        // exit;
         if ($this->input->post('export') == 1) {
             $filename = lang('ogpl') . 'Report';
             $this->load->excel('site/cargo/ogpl-report-list-xls', $data, $filename);
@@ -16859,9 +16865,37 @@
     // https://github.com/JavakarBits/Cargo/issues/207
     public function branch_sales_summary() {
         $this->is_logged_in();
+        $all_zone = $this->api_model->getAllZone();
+        $zone = json_decode($all_zone);
+        $data['zone'] = $zone->data;
 
+        list($stations, $routes) = get_routes_and_stations();
+        $data['stations'] = $stations;
+
+        $all_zone_stations = array();
+        foreach($zone->data as $key=>$zones) {
+            foreach($zones->stations as $k=>$station) {
+                $all_zone_stations[] = $station->code;
+            }
+        }
+        
         $branches = $this->api_model->getOrganization();
         $branches = json_decode($branches);
+
+        $organization = array();
+        foreach ($branches->data as $key=>$branch) {
+            if (in_array($branch->station->code, $all_zone_stations)) {
+                $organization[] = $branch;
+            }
+        }
+
+        usort($organization, function ($a, $b) {
+            return strcasecmp($a->name,  $b->name);
+        });
+        $data['organization'] = $organization;
+        
+        $branches = $this->api_model->getOrganization();
+        $branches = json_decode($branches);
         if (is_array($branches->data)) {
             usort($branches->data, function($a, $b) {
                 return $a->name < $b->name ? -1 : 1;
@@ -16875,15 +16909,19 @@
 
     public function branch_sales_summary_list(){
         $this->is_logged_in();
+        
+        list($fromDate, $toDate) = explode(' | ', $this->input->post('dateRange'));
+        $from_date = date('Y-m-d', strtotime($fromDate));
+        $to_date = date('Y-m-d', strtotime($toDate));
         $data = array(
             'queryCode' => 'RQL3U0474',
-            'fromDate' => $this->input->post('from') ?: 'NA',
-            'toDate' => $this->input->post('to') ?: 'NA',
+            'fromDate' => $from_date ?: 'NA',
+            'toDate' => $to_date ?: 'NA',  
+            'zoneCode' => $this->input->post('zoneCode'),
+            'stationCode' => $this->input->post('stationCode'),
             'organizationCode' => $this->input->post('organizationCode')
         );
-        // if (!has_action_rights('CGO-BR-DUE-ALL-BRAH')) {
-        //     $data['organizationCode'] = $this->session->userdata('org_code');
-        // }
+        
 
         $result = $this->api_model->getDynamicReportDataWithMap($data);
         $result = json_decode($result);
@@ -16890,10 +16928,11 @@
 
         $list = array();
         foreach ($result->data as $row) {
-            $list[$row->booked_at]['booked_at'] = $row->booked_at;
+            $list[$row->booked_at]['booked_at'] =$row->booked_at;
             $list[$row->booked_at]['pickup_handling_amount'] += $row->pickup_handling_amount;
             $list[$row->booked_at]['delivery_handling_amount'] += $row->delivery_handling_amount;
             $list[$row->booked_at]['total_amount'] += $row->total_amount;
+           
             $list[$row->booked_at]['total_handling_amount'] += $row->total_handling_amount;
             $list[$row->booked_at]['total_unload_handling_amount'] += $row->total_unload_handling_amount;
             $list[$row->booked_at]['total_amount_paid'] += $row->total_amount_paid;
@@ -16905,6 +16944,9 @@
             $list[$row->booked_at]['docket_charge'] += $row->docket_charge;
             $list[$row->booked_at]['cargo_count'] += $row->cargo_count;
             $list[$row->booked_at]['total_item_count'] += $row->total_item_count;
+            $list[$row->booked_at]['from_organization_name'] = $row->from_organization_name;
+            $list[$row->booked_at]['round_off_value'] +=$row->round_off_value;
+            // $list[$row->booked_at]['from_organization_name_count'] += count($row->from_organization_name);
             $cgst = $row->cgst > 0 ? $row->service_tax / 2 : 0;
             $sgst = $row->sgst > 0 ? $row->service_tax / 2 : 0;
             $ugst = $row->ugst > 0 ? $row->service_tax / 2 : 0;
@@ -16922,11 +16964,30 @@
         }
 
         ksort($list);
+        // echo "<pre>";
+        // print_r($list);
+        // exit;
         $data['result'] = $list;
         $data['cargo_status'] = config_item('cargo_status');
         $data['cargo_payment_status'] = config_item('cargo_payment_status');
-
+        $data['from'] =$from_date;
+        $data['to']=$to_date;
+        
         if ($this->input->post('export') == 1) {
+            // $this->load->exclude_template();
+            // $content = $this->load->view('site/cargo/branch-sales-summary-list-xls', $data, true);
+            // header('Content-Description: File Transfer');
+            // header('Content-Type: application/octet-stream');
+            // header('Content-Disposition: attachment; filename=Branch-Sales-Summary.xls');
+            // header('Content-Transfer-Encoding: binary');
+            // header('Expires: 0');
+            // header('Cache-Control: must-revalidate');
+            // header('Pragma: public');
+            // header('Content-Length: ' . strlen($content));
+            // ob_clean();
+            // flush();
+            // echo $content;
+            // die;
             $filename = 'Branch-Sales-Summary';
             $this->load->excel('site/cargo/branch-sales-summary-list-xls', $data, $filename);
         } else {
Index: application/views/site/cargo/branch-sales-summary-list-xls.tpl
===================================================================
--- application/views/site/cargo/branch-sales-summary-list-xls.tpl	(revision 5331)
+++ application/views/site/cargo/branch-sales-summary-list-xls.tpl	(working copy)
@@ -3,7 +3,7 @@
         <td colspan="6" align="center"><b>Branch Sales Summary</b></td>
     </tr>
     <tr>
-        <td colspan="3" align="left"><b>From</b>&nbsp;{$smarty.request.from|date_format:$ns_date_format}&nbsp;<b>To</b>&nbsp;{$smarty.request.to|date_format:$ns_date_format}</td>
+        <td colspan="3" align="left"><b>From</b>&nbsp;{$from|date_format:$ns_date_format}&nbsp;<b>To</b>&nbsp;{$to|date_format:$ns_date_format}</td>
         <td colspan="3" align="right"><b>Report At :</b>&nbsp; {$smarty.now|date_format:$ns_datetime_format}</td>
     </tr>
     <tr>
@@ -15,7 +15,11 @@
 <table width="100%">
     <thead>
         <tr>
+            <td align="center">Booking Branch</td>
             <td align="center">Date</td>
+            <td align="center">Booking Branch</td>
+            <td align="center">No Of Lrs</td>
+            <td align="center">No Of Art</td>
             <td align="center">Freight Amount <br>(A)</td>
             <td align="center">Pass Charge <br>(B)</td>
             <td align="center">Loading Charges <br>(C)</td>
@@ -25,6 +29,7 @@
             <td align="center">SGST <br>(G)</td>
             <td align="center">CGST <br>(H)</td>
             <td align="center">IGST <br>(I)</td>
+            <td align="center">Round Off <br>(J)</td>
             <td align="center">Total <br>(F2 = F1+G+H+I)</td>
             <td align="center">Paid</td>
             <td align="center">Topay</td>
@@ -34,16 +39,20 @@
     <tbody>
         {foreach item=row from=$result}
             <tr>
+                <td align="right" >{$row['from_organization_name']}</td>
                 <td align="left" nowrap>{$row['booked_at']|date_format:$ns_date_format}</td>
+                <td align="right" >{$row['cargo_count']|sigma:$t_cargo_amount}</td>
+                <td align="right" >{$row['total_item_count']|sigma:$t_item_count}</td>
                 <td align="right" data-format="{$xlformat['currency']}">{$row['total_amount']|sigma:$t_total_amount}</td>
                 <td align="right" data-format="{$xlformat['currency']}">{$row['pass_charge']|sigma:$t_pass_charge}</td>
                 <td align="right" data-format="{$xlformat['currency']}">{$row['total_handling_amount']|sigma:$t_total_handling_amount}</td>
                 <td align="right" data-format="{$xlformat['currency']}">{$row['total_unload_handling_amount']|sigma:$t_total_unload_handling_amount}</td>
-                <td align="right" data-format="{$xlformat['currency']}">{$row['other_charges_amount']|sigma:$t_other_charges_amount}</td>
+                <td align="right" data-format="{$xlformat['currency']}">{($row['other_charges_amount']+$row['docket_charge'])|sigma:$t_other_charges_amount}</td>
                 <td align="right" data-format="{$xlformat['currency']}">{$row['net_amount']|sigma:$t_net_amount}</td>
                 <td align="right" data-format="{$xlformat['currency']}">{$row['sgst']|sigma:$t_sgst}</td>
                 <td align="right" data-format="{$xlformat['currency']}">{$row['cgst']|sigma:$t_cgst}</td>
                 <td align="right" data-format="{$xlformat['currency']}">{$row['igst']|sigma:$t_igst}</td>
+                <td align="right" data-format="{$xlformat['currency']}">{$row['round_off_value']|sigma:$t_round}</td>
                 <td align="right" data-format="{$xlformat['currency']}">{$row['net_total_amount']|sigma:$t_net_total_amount}</td>
                 <td align="right" data-format="{$xlformat['currency']}">{$row['PAD']|default:0|sigma:$t_PAD}</td>
                 <td align="right" data-format="{$xlformat['currency']}">{$row['TOP']|default:0|sigma:$t_TOP}</td>
@@ -53,7 +62,10 @@
         {if $result|count > 1}
             <tfoot>
                 <tr class="active">
+                    <td align="right"></td>
                     <td align="right">Total</td>
+                    <td align="right" data-format="{$xlformat['currency']}">{$t_cargo_amount}</td>
+                    <td align="right" data-format="{$xlformat['currency']}">{$t_item_count}</td>
                     <td align="right" data-format="{$xlformat['currency']}">{$t_total_amount}</td>
                     <td align="right" data-format="{$xlformat['currency']}">{$t_pass_charge}</td>
                     <td align="right" data-format="{$xlformat['currency']}">{$t_total_handling_amount}</td>
@@ -63,6 +75,7 @@
                     <td align="right" data-format="{$xlformat['currency']}">{$t_sgst}</td>
                     <td align="right" data-format="{$xlformat['currency']}">{$t_cgst}</td>
                     <td align="right" data-format="{$xlformat['currency']}">{$t_igst}</td>
+                    <td align="right" data-format="{$xlformat['currency']}">{$t_round}</td>
                     <td align="right" data-format="{$xlformat['currency']}">{$t_net_total_amount}</td>
                     <td align="right" data-format="{$xlformat['currency']}">{$t_PAD|default:0}</td>
                     <td align="right" data-format="{$xlformat['currency']}">{$t_TOP|default:0}</td>
Index: application/views/site/cargo/branch-sales-summary-list.tpl
===================================================================
--- application/views/site/cargo/branch-sales-summary-list.tpl	(revision 5331)
+++ application/views/site/cargo/branch-sales-summary-list.tpl	(working copy)
@@ -4,7 +4,7 @@
     </tr>
     <tr>
         <td width="50%">
-            <b>From</b>  {$smarty.request.from|date_format:$ns_date_format}  <b>To</b> {$smarty.request.to|date_format:$ns_date_format}
+            <b>From</b>  {$from|date_format:$ns_date_format}  <b>To</b> {$to|date_format:$ns_date_format}
         </td>
         <td width="50%" align="right">
             <span class="bold">Report Date/Time :</span> {date("Y-m-d h:i:s A")}
@@ -16,7 +16,10 @@
 <table id="branch_due_report_table" class="table table-bordered " width="99%">
     <thead>
         <tr class="success">
+            <td align="center">Booking Branch</td>
             <td align="center">Date</td>
+            <td align="center">No Of Lrs</td>
+            <td align="center">No Of Art</td>
             <td align="center">Freight Amount <br>(A)</td>
             <td align="center">Pass Charge <br>(B)</td>
             <td align="center">Loading Charges <br>(C)</td>
@@ -26,6 +29,7 @@
             <td align="center">SGST <br>(G)</td>
             <td align="center">CGST <br>(H)</td>
             <td align="center">IGST <br>(I)</td>
+            <td align="center">Round Off <br>(J)</td>
             <td align="center">Total <br>(F2 = F1+G+H+I)</td>
             <td align="center">Paid</td>
             <td align="center">Topay</td>
@@ -33,18 +37,23 @@
         </tr>
     </thead>
     <tbody>
+    {assign var ="index" value = 0}
         {foreach item=row from=$result}
             <tr>
+                <td align="left" nowrap>{$row['from_organization_name']}</td>
                 <td align="left" nowrap>{$row['booked_at']|date_format:$ns_date_format}</td>
+                <td align="right">{$row['cargo_count']|sigma:$t_cargo_count}</td>
+                <td align="right">{$row['total_item_count']|sigma:$t_item_count}</td>
                 <td align="right">{$row['total_amount']|sigma:$t_total_amount|inr_format}</td>
                 <td align="right">{$row['pass_charge']|sigma:$t_pass_charge|inr_format}</td>
                 <td align="right">{$row['total_handling_amount']|sigma:$t_total_handling_amount|inr_format}</td>
                 <td align="right">{$row['total_unload_handling_amount']|sigma:$t_total_unload_handling_amount|inr_format}</td>
-                <td align="right">{$row['other_charges_amount']|sigma:$t_other_charges_amount|inr_format}</td>
+                <td align="right">{($row['other_charges_amount']+$row['docket_charge'])|sigma:$t_other_charges_amount|inr_format}</td>
                 <td align="right">{$row['net_amount']|sigma:$t_net_amount|inr_format}</td>
                 <td align="right">{$row['sgst']|sigma:$t_sgst|inr_format}</td>
                 <td align="right">{$row['cgst']|sigma:$t_cgst|inr_format}</td>
                 <td align="right">{$row['igst']|sigma:$t_igst|inr_format}</td>
+                <td align="right">{$row['round_off_value']|sigma:$t_round|inr_format}</td>
                 <td align="right">{$row['net_total_amount']|sigma:$t_net_total_amount|inr_format}</td>
                 <td align="right">{$row['PAD']|default:0|sigma:$t_PAD|inr_format}</td>
                 <td align="right">{$row['TOP']|default:0|sigma:$t_TOP|inr_format}</td>
@@ -53,7 +62,10 @@
         {/foreach}
         {if $result|count > 1}
             <tr class="active bold">
+                <td align="right"></td>
                 <td align="right">Total</td>
+                <td align="right">{$t_cargo_count}</td>
+                <td align="right">{$t_item_count}</td>
                 <td align="right">{$t_total_amount|inr_format}</td>
                 <td align="right">{$t_pass_charge|inr_format}</td>
                 <td align="right">{$t_total_handling_amount|inr_format}</td>
@@ -63,6 +75,7 @@
                 <td align="right">{$t_sgst|inr_format}</td>
                 <td align="right">{$t_cgst|inr_format}</td>
                 <td align="right">{$t_igst|inr_format}</td>
+                <td align="right">{$t_round|inr_format}</td>
                 <td align="right">{$t_net_total_amount|inr_format}</td>
                 <td align="right">{$t_PAD|default:0|inr_format}</td>
                 <td align="right">{$t_TOP|default:0|inr_format}</td>
Index: application/views/site/cargo/branch-sales-summary.tpl
===================================================================
--- application/views/site/cargo/branch-sales-summary.tpl	(revision 5331)
+++ application/views/site/cargo/branch-sales-summary.tpl	(working copy)
@@ -1,4 +1,4 @@
-<div class="brand_top">
+    <div class="brand_top">
     <h3>Branch Sales Summary</h3>
     <div class="text-right noprint" id="export-branch-sales" style="display: none">
         <div class="btn-group btn-group-sm">
@@ -27,33 +27,51 @@
                                         <div id="search-panel">
                                             <form id="branch_sales_form" name="branch_sales_form" method="post" onsubmit="return false;">
                                                 <div class="row">
-                                                    <div class="form-group col-md-2">
-                                                        <label for="branch_sales_from">From Date</label>
+                                                    <div class="form-group col-md-4">
+                                                        <label for="branch_sale_date_range" class="req">Date Range</label>
                                                         <div class="input-group">
-                                                            <input type="text" name="from" class="form-control sdate" id="branch_sales_from" readonly="true" placeholder="Date From">
+                                                            <input type="text" name="dateRange" class="form-control sdate" id="branch_sale_date_range" readonly="true" placeholder="Date Range">
                                                             <span class="input-group-addon"><i class="fa fa-calendar"></i></span>
                                                         </div>
                                                     </div>
                                                     <div class="form-group col-md-2">
-                                                        <label for="branch_sales_to">To Date</label>
-                                                        <div class="input-group">
-                                                            <input type="text" name="to" class="form-control sdate" id="branch_sales_to" readonly="true" placeholder="Date To">
-                                                            <span class="input-group-addon"><i class="fa fa-calendar"></i></span>
+                                                        <label for="zone-sales-sumary-report" class="">Zone</label>
+                                                        <div class="input-group col-md-12">
+                                                            <select id="zone-sales-sumary-report" name="zoneCode" class="form-control">
+                                                                <option value="NA" data-stationcode="">All Zone</option>
+                                                                {foreach key=key item=row from=$zone}
+                                                                    {$stationsArray = array()}
+                                                                    {foreach from=$row->stations item=stn key=key}
+                                                                        {$stationsArray[] = $stn->code}
+                                                                    {/foreach}
+                                                                    <option value="{$row->code}" data-stationcode="{implode(",",$stationsArray)}">{$row->name}</option>
+                                                                {/foreach}
+                                                            </select>
                                                         </div>
                                                     </div>
-                                                    {* {if $action_rights['CGO-BR-SALES-ALL-BRAH']>0} *}
+                                                    <div class="col-md-2">
+                                                        <label for="stationCode" class="">Stations</label>
+                                                        <div class="input-group col-md-12">
+                                                            <select id="sales-sumary-station" name="stationCode" class="form-control" onclick="stationfilter();">
+                                                                <option value="NA" data-station="">All Station</option>
+                                                                {foreach item=station from=$stations}
+                                                                    <option value="{$station.code}" data-station="{$station.code}">{$station.name}</option>
+                                                                {/foreach}
+                                                            </select>
+                                                        </div>
+                                                    </div>
                                                     <div class="form-group col-md-2">
-                                                        <label for="branch_sales_organization">Branch</label>
-                                                        <div class="input-group col-md-11">
-                                                            <select id="branch_sales_organization" name="organizationCode" class="form-control">
-                                                                <option value="NA">All Branch</option>
-                                                                {foreach item=row from=$branches}
-                                                                    <option value="{$row->code}">{$row->name}</option>
+                                                        <label for="sales-sumary-branch" class="">Branch</label>
+                                                        <div class="input-group col-md-12">
+                                                            <select id="sales-sumary-branch" name="organizationCode" class="form-control">
+                                                                <option value="NA" data-brachcode="NA">All Branch</option>
+                                                                {foreach item=branch from=$organization key=bcode}
+                                                                    <option value="{$branch->code}" data-brachcode="{$branch->station->code}">{$branch->name}</option>
                                                                 {/foreach}
                                                             </select>
                                                         </div>
                                                     </div>
-                                                    {* {/if} *}
+                                                {* {/if} *}
                                                     <div class="form-group col-md-2">
                                                         <label for="">&nbsp;</label>
                                                         <div class="input-group">
@@ -90,46 +108,113 @@
 </div>
 <iframe name="ifrm_branch_sales" id="ifrm_branch_sales" style="display:none"></iframe>
 
-{literal}
     <script>
-        $('#branch_sales_from').datepicker({
-            //     todayHighlight: true,
-            format: 'yyyy-mm-dd',
-            autoclose: true,
-            startDate: '-3m',
-            endDate: '+3m',
-            changeMonth: true,
-        }).on('changeDate', function (e) {
-            var fDate = new Date(e.date);
-            var end = new Date(fDate.setDate(fDate.getDate() + 31));
-            $('#branch_sales_to').datepicker('setStartDate', e.date);
-            $("#branch_sales_to").datepicker("setEndDate", end)
+        $('#branch_sale_date_range').daterangepicker({
+            buttonClasses: ['btn', 'btn-xs'],
+            format: 'DD-MM-YYYY',
+            separator: ' | ',
+            timePicker: false,
+            ranges: {
+                'Today': [moment().startOf('day'), moment().endOf('day')],
+                'Yesterday': [moment().subtract(1, 'days').startOf('day'), moment().subtract(1, 'days').endOf('day')],
+                'Last 7 Days': [moment().subtract(6, 'days').startOf('day'), moment().endOf('day')],
+                'Last 30 Days': [moment().subtract(29, 'days').startOf('day'), moment().endOf('day')],
+            },
+        }).on('show.daterangepicker', function(ev, picker) {
+            setEndTime(picker);
+            picker.showCalendars();
         });
-        $('#branch_sales_to').datepicker({
-            todayHighlight: true,
-            startDate: '-0m',
-            endDate: '+3m',
-            format: 'yyyy-mm-dd',
-            maxDate: "+365D",
-            autoclose: true
-        }).on('changeDate', function (e) {
-            var fDate = new Date(e.date);
-            var start = new Date(fDate.setDate(fDate.getDate() - 31));
-            $('#branch_sales_from').datepicker('setStartDate', start);
-            $('#branch_sales_from').datepicker('setEndDate', e.date)
+
+        var picker = $('#branch_sale_date_range').data('daterangepicker');
+        picker.container.find('.calendar').on('click', 'td.available', function() {
+            setEndTime(picker);
         });
 
-        $('#branch_sales_organization').select2();
+        function setEndTime(picker) {
+            if (picker.endDate.format('HH:mm:ss') == '00:00:00') {
+                picker.setEndDate(moment(picker.endDate).endOf('day'));
+            }
+        }
 
+        $('#zone-sales-sumary-report, #sales-sumary-branch, #sales-sumary-station').select2();
+        $('#zone-sales-sumary-report, #sales-sumary-branch').val('NA').trigger('change');
+        var zone_branch_list = $('#sales-sumary-branch option');
+        var zone_station_list = $('#sales-sumary-station option')
+        var emptyOrgOption = '<option value="NA" data-brachcode="NA">All Branch</option>';
+        var emptyStnOption = '';
+
+        $('#zone-sales-sumary-report').on('change', function() {
+            $('#sales-sumary-station').val('NA').trigger('change.select2');
+            $('#sales-sumary-branch').val('NA').trigger('change.select2');
+            let zoneValue = $('#zone-sales-sumary-report').val();
+            let stationValue = $('#zone-sales-sumary-report').val();
+            let branchValue = $('#zone-sales-sumary-report').val();
+
+            if ($(this).val() != "" && $(this).val() != 'NA' && stationValue != "NA") {
+                $("#sales-sumary-station").select2("val", "");
+            }
+            if ($(this).val() != "" && $(this).val() != 'NA' && stationValue != "NA" && branchValue != 'NA') {
+                $("#sales-sumary-branch").select2("val", "");
+            }
+
+            var zone_station_code = $('#zone-sales-sumary-report option:selected').data('stationcode');
+            var zone_stn_array = zone_station_code.split(',');
+            branchOfStation(zone_stn_array);
+
+            $('#sales-sumary-station').html(function() {
+            var allOption = $('<option>').val('NA').text('All Station').data('station', "");
+                var filteredOptions = zone_station_list.filter(function() {
+                    return $.inArray($(this).data('station'), zone_stn_array) !== -1 || ($('#zone-sales-sumary-report').val() == 'NA' && $(this).val() != 'NA' && $('#zone-sales-sumary-report').val() != $(this).data('station'));
+                });
+                return [allOption].concat(filteredOptions.toArray());
+            }).val('NA').trigger('change.select2');
+        });
+
+        function stationfilter() {
+            var zone_station_code = $('#sales-sumary-station option:selected').data('station');
+            var zone_stn_array = zone_station_code.split(',');
+            branchOfStation(zone_stn_array);
+            $('#sales-sumary-branch').html(function() {
+                return zone_branch_list.filter(function() {
+                    return $.inArray($(this).data('brachcode'), zone_stn_array) !== -1 || ($('#sales-sumary-station').val() == 'NA' && $(this).val() != 'NA');
+                });
+            }).prepend(emptyOrgOption).val('NA').trigger('change');
+        }
+
+        function branchOfStation(zone_stn_array) {
+            $('#sales-sumary-branch').html(function() {
+                var allOption = $('<option>').val('NA').text('All Branch').data('brachcode', '');
+                var filteredOptions = zone_branch_list.filter(function() {
+                    return $.inArray($(this).data('brachcode'), zone_stn_array) !== -1 || ($('#zone-sales-sumary-report').val() == 'NA' && $(this).val() != 'NA');
+                });
+
+                return [allOption].concat(filteredOptions.toArray());
+            }).val('NA').trigger('change.select2');
+        }
+
+        var user_branch_code = '{$login_station}';
+        if (user_branch_code !== '') {
+            $('#sales-sumary-station').val(user_branch_code).prop('selected', true).change();
+            stationfilter();
+        }
+
         function searchBranchSales() {
             $('#branch-sales-list').html('');
             var data = {};
-            data.from = $('#branch_sales_from').val();
-            data.to = $('#branch_sales_to').val();            
-            data.organizationCode = $('#branch_sales_organization').val();
+            data.dateRange = $('#branch_sale_date_range').val();
+            data.zoneCode = $('#zone-sales-sumary-report').val();
+            data.stationCode = $('#sales-sumary-station').val();
+            data.organizationCode = $('#sales-sumary-branch').val();
 
             var err = 0;
             $('.inp_error').removeClass('inp_error');
+            
+            if (data.dateRange == '') {
+                $('#branch_sale_date_range').addClass('inp_error');
+                err++;
+            } else {
+                $('#branch_sale_date_range').removeClass('inp_error');
+            }
 
             if (err > 0) {
                 $('#branch_sales_state').removeClass('alert-success').addClass('alert-danger').removeClass('hide');
@@ -174,5 +259,4 @@
             document.branch_sales_form.submit();
             return;
         }
-    </script>
-{/literal}
\ No newline at end of file
+    </script>
\ No newline at end of file
