Index: application/controllers/transit.php
===================================================================
--- application/controllers/transit.php	(nonexistent)
+++ application/controllers/transit.php	(working copy)
@@ -0,0 +1,47 @@
+<?php
+
+if (!defined('BASEPATH'))
+    exit('No direct script access allowed');
+
+class transit extends MY_Controller {
+
+    function __construct() {
+        parent::__construct();
+        $this->is_logged_in();
+        $this->_user_hash = _get_user_hash();
+    }
+
+    public function index() {
+        $this->is_logged_in();
+        redirect('user/dashboard');
+    }
+
+    public function ogpl_odometer_comman(){
+        $this->is_logged_in();
+        _include_js('assets/js/common.js');
+        $data = array();
+        $data['login_user'] = $this->session->userdata('user_id');
+        $data['login_branch'] = $this->session->userdata('org_code');
+        $data['loadedVehicle'] =$this->input->post('loadedVehicle');
+        $data['transitCode'] =$this->input->post('transitCode');
+        $transitCode = $this->input->get('transitCode');
+    
+        $odoDetails = $this->api_model->getTransitOdometer(array(
+            'transitCode' => $transitCode
+        ));
+        $odoDetails = json_decode($odoDetails);
+        foreach ($odoDetails->data as $row) {
+            if ($row->organization->code == $data['login_branch']) {
+                $data['odoDetails'] = $row;
+                break;
+            }
+        }
+
+        $vehicle = $this->api_model->get_vehicle_details($this->input->post('loadedVehicle'));
+        $vehicle = json_decode($vehicle);
+        $data['vehicle']=$vehicle->data;
+        $this->load->view('site/odometer-comman', $data);
+    }
+
+    
+}
Index: application/views/site/cargo/add-ogplv3.tpl
===================================================================
--- application/views/site/cargo/add-ogplv3.tpl	(revision 5267)
+++ application/views/site/cargo/add-ogplv3.tpl	(working copy)
@@ -670,14 +670,14 @@
                                             {* setting based strip start module *}
                                             {if $cargo_setting->transitOdometerModelCode == "ORGDTN"}  
                                                 {if $ogpl->fromStation->code == $login_station }
-                                                <button class="btn btn-warning {if !$ogpl || (isset($odoDetails) && $odoDetails->startOdometer > 0) || (!$odoDetails && $show_end_odo == 0)}hide{/if}" id="start-transit-btn" type="button" onclick="showStartTripDialog(true);">Start Trip</button>
+                                                <button class="btn btn-warning {if !$ogpl || (isset($odoDetails) && $odoDetails->startOdometer > 0) || (!$odoDetails && $show_end_odo == 0)}hide{/if}" id="start-transit-btn" type="button" onclick="openOdometerDialog();">Start Trip</button>
                                                {else}
                                                    {if $odoDetails->fromOdometerRange > 0}
-                                                       <button class="btn btn-warning {if !$ogpl || (isset($odoDetails) && $odoDetails->startOdometer > 0) || (!$odoDetails && $show_end_odo == 0) }hide{/if}" id="start-transit-btn" type="button" onclick="showStartTripDialog(true);">Start Trip</button>
+                                                       <button class="btn btn-warning {if !$ogpl || (isset($odoDetails) && $odoDetails->startOdometer > 0) || (!$odoDetails && $show_end_odo == 0) }hide{/if}" id="start-transit-btn" type="button" onclick="openOdometerDialog();">Start Trip</button>
                                                    {/if}
                                                {/if}
                                             {else}
-                                                <button class="btn btn-warning {if !$ogpl || (isset($odoDetails) && $odoDetails->startOdometer > 0) || (!$odoDetails && $show_end_odo == 0)}hide{/if}" id="start-transit-btn" type="button" onclick="showStartTripDialog(true);">Start Trip</button>
+                                                <button class="btn btn-warning {if !$ogpl || (isset($odoDetails) && $odoDetails->startOdometer > 0) || (!$odoDetails && $show_end_odo == 0)}hide{/if}" id="start-transit-btn" type="button" onclick="openOdometerDialog();">Start Trip</button>
                                             {/if}
                                             
 
@@ -843,338 +843,6 @@
     </div>
 </div>
 
-{* start trip *}
-<div id="start-trip-dialog" style="display: none">
-    <div class="clearfix">
-        <a href="javascript:;" class="pull-right" onclick="hideStartTripDialog();"><i class="fa fa-times-circle fa-2x"></i></a>
-    </div>
-    <div id="start-trip-panel" class="clearfix" style="padding: 10px 5px;max-height: 600px;overflow-x: hidden;overflow-y: auto;">
-        <form onsubmit="return false;">
-        <input type="hidden" id="allowLoad">
-            <div class="row">
-                <div class="col-md-4 text-center">
-                    <img src="assets/img/start-trip.png" width="65%">
-                </div>
-                <div class="col-md-4 text-center">
-                    <h4 class="bold">Departure Time</h4>
-                    <h5 id="trip-start-date"></h5>
-                    <h4 id="trip-start-time"></h4>
-                </div>
-                <div class="col-md-4 text-center">
-                    <h5 class="bold">Last odometer Reading</h5>
-                    <h5><span class="last-odometer"></span> Km</h5>
-                </div>
-                <div class="clearfix"></div>
-
-                <div class="col-md-12">
-                    <label class="h5">Fill The Following Details To Start Trip</label>
-                    <h6 class="bold">Odometer</h6>
-                    <div class="row">
-                        <div class="col-md-4">
-                            <label class="control-label req" for="">Odometer</label>
-                            <input type="number" id="start-trip-odo" class="form-control no-spin" placeholder="Odometer">
-                        </div>
-                        <div class="col-md-4 p_l_n">
-                            <div class="input-group col-md-12">
-                                <label class="control-label" for="">Supervisor</label>
-                                <select id="start-trip-supervisor" class="form-control">
-                                    <option value="" selected>Select Supervisor</option>
-                                    {foreach item=row from=$branchContact}
-                                        {if $row->category->code == 'SPVSR'}
-                                            <option value="{$row->code}">{$row->name}</option>                                    
-                                        {/if}
-                                    {/foreach}
-                                </select>
-                            </div>
-                        </div>
-                    </div>
-
-                    <div id="start-trip-expense">
-                        <h6 class="bold">Payments</h6>
-                        <div class="row">
-                        {if  $namespace === "seenutransport" && isset($ownershipTypeing['OWN'])}
-                            <div class="col-md-4">
-                                <label class="control-label" for="">Advance Amount</label>
-                                <input type="number" id="start-trip-advance-amount" class="form-control no-spin" placeholder="Advance Amount">
-                            </div>
-                            <div class="col-md-4">
-                                <div class="input-group col-md-12">
-                                    <label class="control-label" for="">Payment Mode</label>
-                                    <select id="start-trip-adv-paymode" class="form-control" onchange="setStartTripTransactionMode();">
-                                        <option value="" selected>Payment Mode</option>
-                                        {foreach name=o key=k item=name from=$f_t_mode}
-                                            <option value="{$k}">{$name}</option>
-                                        {/foreach} 
-                                    </select>
-                                </div>
-                            </div>
-                            <div class="col-md-4 p_l_n">
-                                <label class="control-label" for="">Payment By</label>
-                                <select id="start-trip-adv-payby" class="form-control">
-                                    <option value="" selected>Payment By</option>
-                                    {foreach from=$contact item=con key=key}
-                                        {foreach from=$con item=val}
-                                            {if $val->contactCategory->actionCode == 'BRCH'}
-                                                <option value="{$val->code}">{$val->name}</option>
-                                            {/if}
-                                        {/foreach}
-                                    {/foreach}
-                                </select>
-                            </div>
-                        {elseif  $namespace === "seenutransport" && isset($ownershipTypeing['HIRE'])}
-                            <div id="start-trip-hiring-amt">
-                                <div class="col-md-4">
-                                    <label class="control-label" for="">Hiring Amount</label>
-                                    <input type="number" class="form-control" id="start-trip-hire-amount" placeholder="Hiring Amount" step="any">
-                                </div>
-                                <div class="clearfix"></div><br>
-                            </div>
-
-                            <div class="col-md-4">
-                                <label class="control-label" for="">Advance Amount</label>
-                                <input type="number" id="start-trip-advance-amount" class="form-control no-spin" placeholder="Advance Amount">
-                            </div>
-                            <div class="col-md-4">
-                                <div class="input-group col-md-12">
-                                    <label class="control-label" for="">Payment Mode</label>
-                                    <select id="start-trip-adv-paymode" class="form-control" onchange="setStartTripTransactionMode();">
-                                        <option value="" selected>Payment Mode</option>
-                                        {foreach name=o key=k item=name from=$f_t_mode}
-                                            <option value="{$k}">{$name}</option>
-                                        {/foreach} 
-                                    </select>
-                                </div>
-                            </div>
-                            <div class="col-md-4 p_l_n">
-                                <label class="control-label" for="">Payment By</label>
-                                <select id="start-trip-adv-payby" class="form-control">
-                                    <option value="" selected>Payment By</option>
-                                    {foreach from=$contact item=con key=key}
-                                        {foreach from=$con item=val}
-                                            {if $val->contactCategory->actionCode == 'BRCH'}
-                                                <option value="{$val->code}">{$val->name}</option>
-                                            {/if}
-                                        {/foreach}
-                                    {/foreach}
-                                </select>
-                            </div>
-                        {elseif  $namespace === "seenutransport" && isset($ownershipTypeing['ATCH'])}
-                            
-                            <div class="col-md-4">
-                                <label class="control-label" for="">Advance Amount</label>
-                                <input type="number" id="start-trip-advance-amount" class="form-control no-spin" placeholder="Advance Amount">
-                            </div>
-                            <div class="col-md-4">
-                                <div class="input-group col-md-12">
-                                    <label class="control-label" for="">Payment Mode</label>
-                                    <select id="start-trip-adv-paymode" class="form-control" onchange="setStartTripTransactionMode();">
-                                        <option value="" selected>Payment Mode</option>
-                                        {foreach name=o key=k item=name from=$f_t_mode}
-                                            <option value="{$k}">{$name}</option>
-                                        {/foreach} 
-                                    </select>
-                                </div>
-                            </div>
-                            <div class="col-md-4 p_l_n">
-                                <label class="control-label" for="">Payment By</label>
-                                <select id="start-trip-adv-payby" class="form-control">
-                                    <option value="" selected>Payment By</option>
-                                    {foreach from=$contact item=con key=key}
-                                        {foreach from=$con item=val}
-                                            {if $val->contactCategory->actionCode == 'BRCH'}
-                                                <option value="{$val->code}">{$val->name}</option>
-                                            {/if}
-                                        {/foreach}
-                                    {/foreach}
-                                </select>
-                            </div>
-                        {else}
-                            <div id="start-trip-hiring-amt">
-                                <div class="col-md-4">
-                                    <label class="control-label" for="">Hiring Amount</label>
-                                    <input type="number" class="form-control" id="start-trip-hire-amount" placeholder="Hiring Amount" step="any">
-                                </div>
-                                <div class="clearfix"></div><br>
-                            </div>
-                            <div class="col-md-4">
-                                <label class="control-label" for="">Advance Amount</label>
-                                <input type="number" id="start-trip-advance-amount" class="form-control no-spin" placeholder="Advance Amount">
-                            </div>
-                            <div class="col-md-4">
-                                <div class="input-group col-md-12">
-                                    <label class="control-label" for="">Payment Mode</label>
-                                    <select id="start-trip-adv-paymode" class="form-control" onchange="setStartTripTransactionMode();">
-                                        <option value="" selected>Payment Mode</option>
-                                        {foreach name=o key=k item=name from=$f_t_mode}
-                                            <option value="{$k}">{$name}</option>
-                                        {/foreach} 
-                                    </select>
-                                </div>
-                            </div>
-                            <div class="col-md-4 p_l_n">
-                                <label class="control-label" for="">Payment By</label>
-                                <select id="start-trip-adv-payby" class="form-control">
-                                    <option value="" selected>Payment By</option>
-                                    {foreach from=$contact item=con key=key}
-                                        {foreach from=$con item=val}
-                                            {if $val->contactCategory->actionCode == 'BRCH'}
-                                                <option value="{$val->code}">{$val->name}</option>
-                                            {/if}
-                                        {/foreach}
-                                    {/foreach}
-                                </select>
-                            </div>
-                        {/if}
-                            
-                            <div class="clearfix"></div><br>
-                            <div class="hide" id="start-trip-adv-upi-mode-panel">
-                                <div class="col-md-4">
-                                    <div class="form-group">
-                                        <label for="start-trip-adv-upi-orgbank">Organization Bank</label>
-                                        <select class="form-control" id="start-trip-adv-upi-orgbank">
-                                            <option value="">Select Organization Bank</option>
-                                            {foreach from=$bankdetail item=val}
-                                                <option value="{$val->code}">{$val->name}, Acc No: {$val->accountNumber}</option>
-                                            {/foreach}
-                                        </select>
-                                    </div>
-                                </div>
-                                <div class="col-md-4">
-                                    <div class="form-group">
-                                        <label for="start-trip-adv-upi-id">UPI ID</label>
-                                        <input type="text" class="form-control" id="start-trip-adv-upi-id" maxlength="250" placeholder="UPI ID" autocomplete="off">
-                                    </div>
-                                </div>
-                                <div class="col-md-4">
-                                    <div class="form-group">
-                                        <label for="start-trip-adv-upi-details">Ref. #</label>
-                                        <input type="text" class="form-control" id="start-trip-adv-upi-details" placeholder="Ref. #" autocomplete="off">
-                                    </div>
-                                </div>
-                            </div>
-
-                            {* Netbanking *}
-                            <div class="hide" id="start-trip-adv-netbanking-mode-panel">
-                                <div class="col-md-6">
-                                    <div class="form-group">
-                                        <label for="start-trip-adv-netbanking-orgbank">Organization Bank</label>
-                                        <select class="form-control" id="start-trip-adv-netbanking-orgbank">
-                                            <option value="">Select Organization Bank</option>
-                                            {foreach from=$bankdetail item=val}
-                                                <option value="{$val->code}">{$val->name}, Acc No: {$val->accountNumber}</option>
-                                            {/foreach}
-                                        </select>
-                                    </div>
-                                </div>
-                                <div class="col-md-6">
-                                    <div class="form-group">
-                                        <label for="start-trip-adv-netbanking-accholder">Account Holder Name</label>
-                                        <input type="text" class="form-control" id="start-trip-adv-netbanking-accholder" maxlength="250" placeholder="Account Holder Name" autocomplete="off">
-                                    </div>
-                                </div>
-                                <div class="col-md-6">
-                                    <div class="form-group" id="start-trip-adv-netbanking-bankname-div">
-                                        <label for="start-trip-adv-netbanking-bankname">Bank Name</label>
-                                        <input type="text" class="form-control" id="start-trip-adv-netbanking-bankname" maxlength="120" placeholder="Bank Name" autocomplete="off">
-                                    </div>
-                                </div>
-                                <div class="col-md-6">
-                                    <div class="form-group">
-                                        <label for="start-trip-adv-netbanking-details">Ref. #</label>
-                                        <input type="text" class="form-control" id="start-trip-adv-netbanking-details" placeholder="Ref. #" autocomplete="off">
-                                    </div>
-                                </div>
-                            </div>
-
-                            {* Cheque *}
-                            <div class="hide" id="start-trip-adv-cheque-mode-panel">
-                                <div class="col-md-3">
-                                    <div class="form-group">
-                                        <label for="start-trip-adv-cheque-payer">Payer Details</label>
-                                        <input type="text" class="form-control" id="start-trip-adv-cheque-payer" maxlength="250" placeholder="Payer Details" autocomplete="off">
-                                    </div>
-                                </div>
-                                <div class="col-md-3">
-                                    <div class="form-group">
-                                        <label for="start-trip-adv-cheque-no">Cheque No</label>
-                                        <input type="text" class="form-control" id="start-trip-adv-cheque-no" maxlength="60" placeholder="Cheque No" autocomplete="off">
-                                    </div>
-                                </div>
-                                <div class="col-md-3">
-                                    <div class="form-group">
-                                        <label for="start-trip-adv-cheque-date">Cheque Date</label>
-                                        <input type="text" class="form-control" id="start-trip-adv-cheque-date" maxlength="20" placeholder="Cheque Date" autocomplete="off" readonly>
-                                    </div>
-                                </div>
-                                <div class="col-md-3">
-                                    <div class="form-group">
-                                        <label for="start-trip-adv-cheque-bankcity">Bank City</label>
-                                        <input type="text" class="form-control" id="start-trip-adv-cheque-bankcity" maxlength="90" placeholder="Bank City" autocomplete="off">
-                                    </div>
-                                </div>
-                                <div class="col-md-6">
-                                    <div class="form-group" id="start-trip-adv-cheque-bankname-div">
-                                        <label for="start-trip-adv-cheque-bankname">Bank Name</label>
-                                        <input type="text" class="form-control" id="start-trip-adv-cheque-bankname" maxlength="120" placeholder="Bank Name" autocomplete="off">
-                                    </div>
-                                </div>
-                                <div class="col-md-6">
-                                    <div class="form-group">
-                                        <label for="start-trip-adv-cheque-remarks">Remarks</label>
-                                        <input type="text" class="form-control" placeholder="Remarks" id="start-trip-adv-cheque-remarks" data-mask="englishonly"/>
-                                    </div>
-                                </div>
-                            </div>
-                        </div>
-
-                        <h6 class="bold">Fuel Expense</h6>
-                        <div class="row">
-                            <div class="col-md-2">
-                                <label class="control-label" for="">Fuel Litres</label>
-                                <input type="number" id="start-trip-fuel-litres" class="form-control no-spin" placeholder="Fuel Litres">
-                            </div>
-                            <div class="col-md-2">
-                                <label class="control-label" for="">Fuel Rate / Liter</label>
-                                <input type="number" class="form-control" id="start-trip-fuel-rate" placeholder="Fuel Rate / Liter" step="any">
-                            </div>
-                            <div class="col-md-3">
-                                <div class="input-group col-md-12">
-                                    <label class="control-label" for="">Vendor</label>
-                                    <select id="start-trip-fuel-vendor" class="form-control">
-                                        <option value="" selected>Select Vendor</option>
-                                        {foreach item=row from=$vendors} 
-                                            {if $row->contactCategory->actionCode == 'RAC'}
-                                                <option value="{$row->code}">{$row->name}</option> 
-                                            {/if}
-                                        {/foreach}
-                                    </select>
-                                </div>
-                            </div>
-                            <div class="col-md-3">
-                                <div class="input-group col-md-12">
-                                    <label for="start-trip-fuel-payment-mode" class="req">Payment Mode</label>
-                                    <select id="start-trip-fuel-payment-mode" class="form-control">
-                                        <option value="CASH">Cash Payment</option> 
-                                        <option value="CRDT">Credit Payment</option> 
-                                    </select>
-                                </div>
-                            </div>
-                            <div class="col-md-2">
-                                <label class="control-label" for="">Fuel Coupon</label>
-                                <input type="text" class="form-control" id="start-trip-fuel-coupon" placeholder="Fuel Coupon">
-                            </div>
-                        </div>
-                    </div>
-                </div>
-                <div class="col-md-12 text-center"><br>
-                    <div id="start-trip-action-state"></div>
-                    <button type="submit" class="btn btn-warning col-md-3" id="start-trip-btn" onclick="startTransitV3Trip();" style="float: none;">Start Trip</button>
-                </div>
-            </div>
-        </form>
-    </div>
-</div>
-
 <script id="vehicle-select-tpl" type="text/html">
     <div class="row">
         <div class="col-xs-7">
@@ -1330,8 +998,8 @@
     var splittedLRList = [];
 
     function checkOdoUpdation(){
-        if(startOdoUpdated === 0){
-            showStartTripDialog(true);
+        if(odoDetails.startOdometer == 0){
+            openOdometerDialog();
             return false;
         }else{
             true;
@@ -2768,7 +2436,7 @@
                             if(loadtype == 'EMPTY_LOAD' && transType == 'INTRNT' ){ 
                                 showStartTripDialog(false); //don't allow to direct or refres the page.
                              }else{
-                                showStartTripDialog(true); //allow to redirect
+                                openOdometerDialog(); //allow to redirect
                              }
                         } else {
                             loading_overlay.update({
@@ -4946,10 +4614,13 @@
         // Out For Delivery End
         //important
         function saveTransitV3(action) {
+            if(odoDetails.startOdometer == 0){
+                openOdometerDialog();
+            }
             var transitList = [];
-                $('#transit-type [name="transit-type"]:checked').each(function() {
-                    transitList.push($(this).val());
-                });
+            $('#transit-type [name="transit-type"]:checked').each(function() {
+                transitList.push($(this).val());
+            });
             var is_emtpy_with_load = transitList.includes('INTRNT') &&  transitList.includes('OTFD');
             var onlyEmptyLoad = transitList.includes('INTRNT') && !transitList.includes('OTFD');
             var activityType = [];
@@ -4985,6 +4656,7 @@
                     url: base_url + 'cargo/save-transitv3',
                     data: data,
                     success: function (response) {
+                       
                         $('#transit-complete-btn').prop('disabled', false);
                         if (response.status == 1) {
                             var errors = [];
@@ -5077,301 +4749,6 @@
             $('#start-trip-dialog').hide();
         }
 
-        function startTransitV3Trip() {
-            var allowLoading = $('#allowLoad').val();
-            if(allowLoading == true){
-            }else{
-                alert('This OGPL vehicle cannot ablicable to load LRs');
-            }
-            var data = {}, err = 0;
-            data.action = 'DEPARTURE';
-            data.transitCode = OGPLFormData.code;
-            data.transitCargo = {};
-            data.transitCargo.code = OGPLFormData.code;
-            data.fromOrganization = {};
-            data.fromOrganization.code = user_login_branch;
-            data.startOdometer = Number($('#start-trip-odo').val());
-            data.departureAt = moment().format('YYYY-MM-DD HH:mm:ss');
-            data.fromOrganizationContact = {};
-            data.fromOrganizationContact.code = $('#start-trip-supervisor').val();
-
-            $('.inp_error').removeClass('inp_error');
-            if (isNaN(data.startOdometer) || data.startOdometer < 0 || data.startOdometer == '') {
-                $('#start-trip-odo').addClass('inp_error');
-                err++;
-            }
-            var vehicle = _.find(OGPLVehicles, o => o.code == OGPLFormData.busVehicle.code) || {};
-            if (vehicle && vehicle.lastOdometer > 0 && data.startOdometer < vehicle.lastOdometer) {
-                $('#start-trip-action-state').removeClass('alert-success hide').addClass('alert alert-danger').show();
-                $('#start-trip-action-state').html('Entered odometer value should be greater than last odometer value');
-                return;
-            }
-
-            if (isNull(odoDetails) || (odoDetails && (odoDetails.startOdometer == 0))) {
-                // fuel expense
-                data.fuelExpenseCode = '';
-                data.activeFlag = 1;
-                data.fuelDate = moment().format('YYYY-MM-DD');
-                data.vehicle = {};
-                data.vehicle.code = OGPLFormData.busVehicle.code;
-                
-                data.vendorContact = {};
-                data.vendorContact.code = $('#start-trip-fuel-vendor').val();
-
-                data.transactionMode = {};
-                data.transactionMode.code = $('#start-trip-fuel-payment-mode').val();
-
-                data.litres = Number($('#start-trip-fuel-litres').val());
-                data.pricePerLitre = Number($('#start-trip-fuel-rate').val());
-                data.totalAmount = data.litres * data.pricePerLitre;
-                data.billNumber = $.trim($('#start-trip-fuel-coupon').val());
-
-                // hiring amount and advance amount
-                data.ogpl = [];
-
-                if ((cargoSetting.userAccountModel.code == 'TXRT' && vehicle.ownershipType.code == "HIRE") || cargoSetting.userAccountModel.code != 'TXRT') {
-                    var hire = {};
-
-                    hire.code = '';
-                    hire.activeFlag = 1;
-                    hire.amount = Number($('#start-trip-hire-amount').val());
-                    hire.remarks = '';
-
-                    hire.user = {};
-                    hire.user.code = login_user;
-
-                    hire.transitCargo = {};
-                    hire.transitCargo.code = OGPLFormData.code;
-                    
-                    hire.cashbookType = {};
-                    hire.cashbookType.code = 'HIRING';
-
-                    if (Number(hire.amount) > 0) {
-                        data.ogpl.push(hire);
-                    }
-                }
-                
-                var advance = {};
-
-                advance.code = '';
-                advance.activeFlag = 1;
-                advance.amount = Number($('#start-trip-advance-amount').val());
-                advance.remarks = '';
-
-                advance.user = {};
-                advance.user.code = login_user;
-
-                advance.transitCargo = {};
-                advance.transitCargo.code = OGPLFormData.code;
-
-                advance.transactionMode = {};
-                advance.transactionMode.code = $('#start-trip-adv-paymode').val();
-                
-                advance.cashbookType = {};
-                advance.cashbookType.code = 'ADVANCE';
-
-                advance.paymentByContact = {};
-                advance.paymentByContact.code = $('#start-trip-adv-payby').val();
-
-                if (advance.transactionMode.code == 'UPI') {
-                    advance.bankDetails = {};
-                    advance.bankDetails.code = $('#start-trip-adv-upi-orgbank').val();
-                    advance.addAttr1 = $.trim($('#start-trip-adv-upi-id').val());
-                    advance.addAttr2 = $.trim($('#start-trip-adv-upi-details').val());
-                } else if (advance.transactionMode.code == 'NBK') {
-                    advance.bankDetails = {};
-                    advance.bankDetails.code = $('#start-trip-adv-netbanking-orgbank').val();
-                    advance.addAttr1 = $.trim($('#start-trip-adv-netbanking-accholder').val());
-                    advance.addAttr2 = $.trim($('#start-trip-adv-netbanking-bankname').val());
-                    advance.addAttr3 = $.trim($('#start-trip-adv-netbanking-details').val());
-                } else if (advance.transactionMode.code == 'CHEQUE') {
-                    advance.chequeDetails = {};
-                    advance.chequeDetails.code = $('#start-trip-adv-chq-code').val();
-                    advance.chequeDetails.activeFlag = 1;
-
-                    advance.chequeDetails.organization = {};
-                    advance.chequeDetails.organization.code = '{$login_branch}';
-
-                    advance.chequeDetails.bankDetails = $.trim($('#start-trip-adv-cheque-bankname').val());
-                    advance.chequeDetails.payerDetails = $.trim($('#start-trip-adv-cheque-payer').val());
-
-                    advance.chequeDetails.chequeDate = moment($('#start-trip-adv-cheque-date').val(), 'DD-MM-YYYY').format('YYYY-MM-DD');
-                    advance.chequeDetails.chequeNo = $.trim($('#start-trip-adv-cheque-no').val());
-                    advance.chequeDetails.bankCity = $.trim($('#start-trip-adv-cheque-bankcity').val());
-
-                    advance.chequeDetails.chequeStatus = {};
-                    advance.chequeDetails.chequeStatus.code = 'NEW';
-
-                    advance.chequeDetails.remarks = $.trim($('#start-trip-adv-cheque-remarks').val());
-                }
-
-                if (Number(advance.amount) > 0) {
-                    data.ogpl.push(advance);
-                }
-
-                // advance amount
-                if ((cargoSetting.userAccountModel.code == 'TXRT' && $.inArray(OGPLFormData.cargoActivityType.code, ['OTFD', 'INTRNT']) != -1 && vehicle.ownershipType.code == "OWN" && !isNull(hire) && hire.amount != '') || (cargoSetting.userAccountModel.code != 'TXRT' && !isNull(hire) && hire.amount != '') || advance.amount != '' || advance.transactionMode.code != '' || advance.paymentByContact.code != '') {
-                    if (!isNull(hire) && hire.amount == '') {
-                        $('#start-trip-hire-amount').addClass('inp_error');
-                        err++;
-                    }
-                    
-                    if (advance.amount == '') {
-                        $('#start-trip-advance-amount').addClass('inp_error');
-                        err++;
-                    }
-
-                    if (advance.transactionMode.code == '') {
-                        $('#start-trip-adv-paymode').addClass('inp_error');
-                        err++;
-                    }
-
-                    if (advance.paymentByContact.code == '') {
-                        $('#start-trip-adv-payby').addClass('inp_error');
-                        err++;
-                    }
-
-                    if (advance.transactionMode.code == 'UPI') {
-                        if (advance.addAttr1 == '') {
-                            $('#start-trip-adv-upi-id').addClass('inp_error');
-                            err++;
-                        }
-
-                        if ($('#start-trip-adv-upi-orgbank').val() == '') {
-                            $('#start-trip-adv-upi-orgbank').addClass('inp_error');
-                            err++;
-                        }
-
-                        if (advance.addAttr2 == '') {
-                            $('#start-trip-adv-upi-details').addClass('inp_error');
-                            err++;
-                        }
-                    } else if (advance.transactionMode.code == 'NBK') {
-                        if (advance.addAttr1 == '') {
-                            $('#start-trip-adv-netbanking-accholder').addClass('inp_error');
-                            err++;
-                        }
-
-                        if (advance.addAttr2 == '') {
-                            $('#start-trip-adv-netbanking-bankname').addClass('inp_error');
-                            err++;
-                        }
-
-                        if ($('#start-trip-adv-netbanking-orgbank').val() == '') {
-                            $('#start-trip-adv-netbanking-orgbank').addClass('inp_error');
-                            err++;
-                        }
-
-                        if (advance.addAttr3 == '') {
-                            $('#start-trip-adv-netbanking-details').addClass('inp_error');
-                            err++;
-                        }
-                    } else if (advance.transactionMode.code == 'CHEQUE') {
-                        if (advance.chequeDetails.payerDetails == '') {
-                            $('#start-trip-adv-cheque-payer').addClass('inp_error');
-                            err++;
-                        }
-
-                        if ($('#start-trip-adv-cheque-date').val() == '') {
-                            $('#start-trip-adv-cheque-date').addClass('inp_error');
-                            err++;
-                        }
-
-                        if (advance.chequeDetails.chequeNo == '') {
-                            $('#start-trip-adv-cheque-no').addClass('inp_error');
-                            err++;
-                        }
-
-                        if (advance.chequeDetails.bankDetails == '') {
-                            $('#start-trip-adv-cheque-bankname').addClass('inp_error');
-                            err++;
-                        }
-
-                        if (advance.chequeDetails.bankCity == '') {
-                            $('#start-trip-adv-cheque-bankcity').addClass('inp_error');
-                            err++;
-                        }
-                    }
-                }
-
-                // fuel expense
-                if (data.vendorContact.code != '' || data.litres != '' || data.pricePerLitre != '' || data.billNumber != '') {
-                    if (data.vendorContact.code == '') {
-                        $('#start-trip-fuel-vendor').addClass('inp_error');
-                        err++;
-                    }
-
-                    if (data.litres == '') {
-                        $('#start-trip-fuel-litres').addClass('inp_error');
-                        err++;
-                    }
-
-                    if (data.pricePerLitre == '') {
-                        $('#start-trip-fuel-rate').addClass('inp_error');
-                        err++;
-                    }
-
-                    if (data.billNumber == '') {
-                        $('#start-trip-fuel-coupon').addClass('inp_error');
-                        err++;
-                    }
-                }
-            }
-            
-            if(err) {
-                $('#start-trip-action-state').removeClass('alert-success hide').addClass('alert alert-danger').show();
-                $('#start-trip-action-state').html('Please enter/select the values in the field that are marked in red');
-                return;
-            }
-            $('#start-trip-panel-btn').hide();
-            $('#start-trip-action-state').removeClass('alert').removeClass('alert-danger').removeClass('alert-success').html(loading_popup);
-            var transitList = [];
-                $('#transit-type [name="transit-type"]:checked').each(function() {
-                    transitList.push($(this).val());
-                });
-            //check is empty load with outfordelivery    
-            var is_emtpy_with_load = transitList.includes('INTRNT') &&  transitList.includes('OTFD');
-            var onlyEmptyLoad = transitList.includes('INTRNT') && !transitList.includes('OTFD');
-            // if(is_emtpy_with_load){
-            //     console.log($('.local-via-branches').val());
-            //     let toOrganizationForLocal=$('#local-via-branches option:selected').val();
-            //         data.toOrganization.code=toOrganizationForLocal;
-            // }
-            $.ajax({
-                type: "POST",
-                dataType: 'json',
-                url: base_url + "cargo/update-start-odometer",
-                data: data,
-                success: function (response) {
-                    if (response.status == 1) {       
-                        startOdoUpdated == 1;         
-                        $('#start-trip-action-state').addClass('alert').addClass('alert-success').html('Your request processed successfully');
-                        window.setTimeout(function () {
-                            hideStartTripDialog();
-                            if(!is_emtpy_with_load){
-                                
-                                let transitcode=OGPLFormData.code;
-                                var params = {};
-                                params.transitCode = transitcode;
-                                    
-                                    if(allowLoading == false){
-                                        window.location.reload();
-                                    }else{
-                                        var url = base_url + "#cargo/add-ogpl?" + $.param(params)+'&INTRNT=HIDE';
-                                        window.open(url, "_self");
-                                        checkURL();
-                                    }
-                            }
-                        }, 3000);
-                    } else {
-                        $('#start-trip-action-state').addClass('alert').addClass('alert-danger').html(response.errorDesc);
-                        $('#start-trip-panel-btn').show();
-                    }
-
-                }
-            });
-        }
-
         function showLoadedLrDialog() {
             $('#transitv3-loaded-lr-dialog').css('background', '#fff');
             $('#transitv3-loaded-lr-dialog').removeClass('hide');
@@ -6515,7 +5892,10 @@
         }
     });
 }
-  
+  function  openOdometerDialog(){
+    var loadVehicle =$('#ogpl-search-vehicle').select2('data');
+    getOdometter(loadVehicle.code , transitOgplcode);
+  }
 </script>
 
 {/literal}
Index: application/views/site/cargo/booking-details.tpl
===================================================================
--- application/views/site/cargo/booking-details.tpl	(revision 5267)
+++ application/views/site/cargo/booking-details.tpl	(working copy)
@@ -426,7 +426,7 @@
                                 <span class="pull-right">
                                     {foreach from=$summary->transitCargoDetails name=tcd item=transitCargoDetails}
                                         {if $smarty.foreach.tcd.last}
-                                        <a href="javascript:;" onclick="print_cargo_ogpl_chart('{$transitCargoDetails->code}');">view {lang('ogpl')}</a>
+                                        {* <a href="javascript:;" onclick="print_cargo_ogpl_chart('{$transitCargoDetails->code}');">view {lang('ogpl')}</a> *}
                                         {/if}
                                     {/foreach}
                                 </span>
Index: application/views/site/cargo/ogpl-chart-cityparcel.tpl
===================================================================
--- application/views/site/cargo/ogpl-chart-cityparcel.tpl	(revision 5267)
+++ application/views/site/cargo/ogpl-chart-cityparcel.tpl	(working copy)
@@ -338,7 +338,14 @@
 
 <script>
     var transitCode = {$ogpl|json_encode};
+    var OGPLVehicles = {$vehicles|json_encode};
     var namespaceCode = {$namespaceCode|json_encode};
+    var ogpl={$ogpl|json_encode};
+    var odoDetails={$odoDetails|json_encode};
+    var transitOgplcode='{$smarty.request.transitCode}';
+    var user_login_branch = {$login_branch|json_encode};
+    var cargoSetting = {$cargo_setting|json_encode};
+    var startOdoUpdated = odoDetails != null && odoDetails.startOdometer != null ? odoDetails.startOdometer : 0;
     $('[data-toggle="tooltip"]').tooltip({
         html: true
     });
@@ -359,6 +366,9 @@
     });
 
     function print_cargo_ogpl_chart(code, groupby) {
+        if(checkOdoUpdation()){
+            return;
+        }
         var route = $('#ogpl-chart-print').val();
         var param = $.param({
             transitCode: code,
@@ -402,4 +412,14 @@
             window.open(url, '_blank');
         }
     }
+
+    function checkOdoUpdation(){
+        if(odoDetails.startOdometer == 0){
+            getOdometter(ogpl.busVehicle.code , transitOgplcode);
+            return true;
+        }else{
+            false;
+        }
+   
+    }
 </script>
\ No newline at end of file
Index: application/views/site/cargo/ogpl-chart.tpl
===================================================================
--- application/views/site/cargo/ogpl-chart.tpl	(revision 5267)
+++ application/views/site/cargo/ogpl-chart.tpl	(working copy)
@@ -446,8 +446,15 @@
     </div>
 </div>
 
-<script>
+<script> 
+    var OGPLVehicles = {$vehicles|json_encode};
+    var ogpl={$ogpl|json_encode};
+    var odoDetails={$odoDetails|json_encode};
+    var transitOgplcode='{$smarty.request.transitCode}';
+    var user_login_branch = {$login_branch|json_encode};
     var cargoSetting = {$cargo_setting|json_encode};
+    var startOdoUpdated = odoDetails != null && odoDetails.startOdometer != null ? odoDetails.startOdometer : 0;
+
     $('[data-toggle="tooltip"]').tooltip({
         html: true
     });
@@ -477,6 +484,9 @@
     }
 
     function print_cargo_ogpl_chart(code, groupby) {
+        if(checkOdoUpdation()){
+            return;
+        }
         var route = $('#ogpl-chart-print').val();
         var param = $.param({
             transitCode: code,
@@ -488,4 +498,14 @@
         var wo = window.open(url, "_blank", "toolbar=no, scrollbars=yes, resizable=yes, top=50, left=50, width=1000, height=650");
         wo.print();
     }
+
+    function checkOdoUpdation(){
+        if(odoDetails.startOdometer == 0){
+            getOdometter(ogpl.busVehicle.code , transitOgplcode);
+            return true;
+        }else{
+            false;
+        }
+       
+    }
 </script>
\ No newline at end of file
Index: assets/js/common.js
===================================================================
--- assets/js/common.js	(revision 5267)
+++ assets/js/common.js	(working copy)
@@ -1,134 +1,193 @@
 // for debugging start
-function lineWithArg(argName=null){
-    var line='';
-        if(argName!=null){
-        line=`-----------------------------${argName}-------------------------------------`;
-        }else{
-        line=`-----------------------------------------------------------------------------`;
-        }
-    console.log(line);
+function lineWithArg(argName = null) {
+  var line = "";
+  if (argName != null) {
+    line = `-----------------------------${argName}-------------------------------------`;
+  } else {
+    line = `-----------------------------------------------------------------------------`;
+  }
+  console.log(line);
 }
 
-function printJson(data,argName=null){
-    if(allowPrint === true){
-        alert(allowPrint);
-        lineWithArg(argName);
-        console.log(JSON.stringify(data, null, 3));
-    }
+function printJson(data, argName = null) {
+  if (allowPrint === true) {
+    alert(allowPrint);
+    lineWithArg(argName);
+    console.log(JSON.stringify(data, null, 3));
+  }
 }
 
-function print_r(data,argName=null){
-    if(allowPrint === true){
-        lineWithArg(argName);
-        console.log(data);
-    }
+function print_r(data, argName = null) {
+  if (allowPrint === true) {
+    lineWithArg(argName);
+    console.log(data);
+  }
 }
 // for debugging end
 
 //export Excel Function  ex: exportExcel('table_id_name', 'excel file name');
 var exportExcel = (function () {
-    var uri = 'data:application/vnd.ms-excel;base64,';
-    var template = '<html xmlns:o="urn:schemas-microsoft-com:office:office" xmlns:x="urn:schemas-microsoft-com:office:excel" xmlns="http://www.w3.org/TR/REC-html40"><head>' +
-        '<style>th{  color: #fff; padding: 10px; } thead{ background-color: #64b92a; }</style>' +
-        '<!--[if gte mso 9]><xml><x:ExcelWorkbook><x:ExcelWorksheets><x:ExcelWorksheet>' +
-        '<x:Name>{worksheet}</x:Name><x:WorksheetOptions><x:DisplayGridlines/></x:WorksheetOptions>' +
-        '</x:ExcelWorksheet></x:ExcelWorksheets></x:ExcelWorkbook></xml><![endif]-->' +
-        '<meta http-equiv="content-type" content="text/plain; charset=UTF-8"/></head><body><table style="font-weight:500;" border="1px" cellpadding="0" cellspacing="0" >{table}</table></body></html>';
+  var uri = "data:application/vnd.ms-excel;base64,";
+  var template =
+    '<html xmlns:o="urn:schemas-microsoft-com:office:office" xmlns:x="urn:schemas-microsoft-com:office:excel" xmlns="http://www.w3.org/TR/REC-html40"><head>' +
+    "<style>th{  color: #fff; padding: 10px; } thead{ background-color: #64b92a; }</style>" +
+    "<!--[if gte mso 9]><xml><x:ExcelWorkbook><x:ExcelWorksheets><x:ExcelWorksheet>" +
+    "<x:Name>{worksheet}</x:Name><x:WorksheetOptions><x:DisplayGridlines/></x:WorksheetOptions>" +
+    "</x:ExcelWorksheet></x:ExcelWorksheets></x:ExcelWorkbook></xml><![endif]-->" +
+    '<meta http-equiv="content-type" content="text/plain; charset=UTF-8"/></head><body><table style="font-weight:500;" border="1px" cellpadding="0" cellspacing="0" >{table}</table></body></html>';
 
-        return function (tableId, name) {
-        var table = document.getElementById(tableId);
-        if (!table) {
-            return;
-        }
-        
-        var blob = new Blob([template.replace('{worksheet}', name || 'Worksheet').replace('{table}', table.innerHTML)], {
-            type: 'application/vnd.ms-excel'
-        });
-    
-        var link = document.createElement('a');
-        link.href = window.URL.createObjectURL(blob);
-        link.download = name + '.xls';
-        link.click();
-    };
+  return function (tableId, name) {
+    var table = document.getElementById(tableId);
+    if (!table) {
+      return;
+    }
+
+    var blob = new Blob(
+      [
+        template
+          .replace("{worksheet}", name || "Worksheet")
+          .replace("{table}", table.innerHTML),
+      ],
+      {
+        type: "application/vnd.ms-excel",
+      }
+    );
+
+    var link = document.createElement("a");
+    link.href = window.URL.createObjectURL(blob);
+    link.download = name + ".xls";
+    link.click();
+  };
 })();
 // export function end
 
-//Print Element Function 
-function printTable(elm,name) {
-    // console.log(elm);
-    let element=$("#"+elm+"");
-    element.css('border', '1px solid black');
-    element.printThis({
-                importCSS: true,
-                importStyle: true,
-                loadCSS: "assets/bootstrap/css/bootstrap.min.css",
-                header: "<h1>"+name+"</h1>",
-                beforePrintEvent:element.css('border', '1px solid black'),
-                afterPrintEvent: element.css('border', '1px solid black')
-            });
+//Print Element Function
+function printTable(elm, name) {
+  // console.log(elm);
+  let element = $("#" + elm + "");
+  element.css("border", "1px solid black");
+  element.printThis({
+    importCSS: true,
+    importStyle: true,
+    loadCSS: "assets/bootstrap/css/bootstrap.min.css",
+    header: "<h1>" + name + "</h1>",
+    beforePrintEvent: element.css("border", "1px solid black"),
+    afterPrintEvent: element.css("border", "1px solid black"),
+  });
 } //example :  printTable('table_id','file name')
 //Print Element function end
 
 //get intersection values in an array (for payment types task)
 function findIntersection(array1, array2) {
-    return array1.filter(obj1 =>
-        array2.some(obj2 =>
-        Object.keys(obj1).every(key =>
-            obj2.hasOwnProperty(key) && obj1[key] === obj2[key]
-        )
-        )
-    );
+  return array1.filter((obj1) =>
+    array2.some((obj2) =>
+      Object.keys(obj1).every(
+        (key) => obj2.hasOwnProperty(key) && obj1[key] === obj2[key]
+      )
+    )
+  );
 }
 
-
 //show Success or error message in page
 var alertMsg = (function () {
-    function successMsg(className, message, title = '') {
-        var targetElement = $('.' + className);
-            var titleContent = title !=""?`<strong>${title}</strong>`:'<strong>Success !</strong>';
-            var html = `<div class="alert alert-success alert-dismissible show" w-100 role="alert">
+  function successMsg(className, message, title = "") {
+    var targetElement = $("." + className);
+    var titleContent =
+      title != "" ? `<strong>${title}</strong>` : "<strong>Success !</strong>";
+    var html = `<div class="alert alert-success alert-dismissible show" w-100 role="alert">
                              ${titleContent} ${message}
                             <button type="button" class="btn-close" data-bs-dismiss="alert" aria-label="Close"></button>
                         </div>`;
-            targetElement.html(html);
-            if(alertClosingtime !=false){
-                setTimeout(() => {
-                    targetElement.html('');
-                }, alertClosingtime);
-            }
+    targetElement.html(html);
+    if (alertClosingtime != false) {
+      setTimeout(() => {
+        targetElement.html("");
+      }, alertClosingtime);
     }
+  }
 
-    function errorMsg(className, message, title = '') {
-        var targetElement = $('.' + className);
-        var titleContent = title !=""?`<strong>${title}</strong>`:'<strong>Attention !</strong>';
-            var html = `<div class="alert alert-danger alert-dismissible w-100 show" role="alert">
+  function errorMsg(className, message, title = "") {
+    var targetElement = $("." + className);
+    var titleContent =
+      title != ""
+        ? `<strong>${title}</strong>`
+        : "<strong>Attention !</strong>";
+    var html = `<div class="alert alert-danger alert-dismissible w-100 show" role="alert">
                            ${titleContent} ${message}
                             <button type="button" class="btn-close" data-bs-dismiss="alert" aria-label="Close"></button>
                         </div>`;
-            targetElement.html(html);
-            if(alertClosingtime !=false){
-                setTimeout(() => {
-                    targetElement.html('');
-                }, alertClosingtime);
-            }
+    targetElement.html(html);
+    if (alertClosingtime != false) {
+      setTimeout(() => {
+        targetElement.html("");
+      }, alertClosingtime);
     }
+  }
 
-    function hideMsg(className) {
-        var targetElement = $('.' + className);
-        targetElement.html('');
-    }
+  function hideMsg(className) {
+    var targetElement = $("." + className);
+    targetElement.html("");
+  }
 
-    return {
-        successMsg: successMsg,
-        errorMsg: errorMsg,
-        hideMsg: hideMsg
-    };
+  return {
+    successMsg: successMsg,
+    errorMsg: errorMsg,
+    hideMsg: hideMsg,
+  };
 })();
 
-//btn blocking 
-function btnBlock(element){
-    $(element).attr('disabled',true);
-    setInterval(() => {
-        $(element).attr("disabled", false);
-    },releaseTime);
-}
\ No newline at end of file
+//btn blocking
+function btnBlock(element) {
+  $(element).attr("disabled", true);
+  setInterval(() => {
+    $(element).attr("disabled", false);
+  }, releaseTime);
+}
+
+function getOdometter(vehicleCode, transitCode) {
+    var data = {};
+    data.loadedVehicle = vehicleCode;
+    data.transitCode = transitCode;
+    $.ajax({
+      type: "POST",
+      url: base_url + "transit/ogpl-odometer-comman",
+      data: data,
+      success: function (response) {
+        $("#cargo-invoice-iframe").html(response);
+        $("#cargo-invoice-pdf-dialog").css("background", "#fff");
+        $("#cargo-invoice-pdf-dialog").removeClass("hide");
+        $("#cargo-invoice-pdf-dialog").dialog({
+          autoOpen: true,
+          height: 700,
+          width: 850,
+          modal: true,
+          resizable: false,
+          closeOnEscape: false,
+        });
+        $("#start-trip-dialog").show();
+        $("#start-trip-dialog .inp_error").removeClass("inp_error");
+        $("#start-trip-action-state")
+          .removeClass("alert alert-danger alert-success")
+          .html("");
+        $(
+          "#start-trip-odo, #start-trip-hire-amount, #start-trip-advance-amount, #start-trip-adv-paymode, #start-trip-fuel-litres, #start-trip-fuel-rate, #start-trip-fuel-coupon"
+        ).val("");
+        $("#start-trip-supervisor, #start-trip-fuel-vendor, #start-trip-adv-payby")
+          .val("")
+          .trigger("change");
+        $("#start-trip-fuel-payment-mode").val("CASH").trigger("change");
+        $("#start-trip-hiring-amt").removeClass("hide");
+        $("#allowLoad").val(true);
+
+        $("#trip-start-date").html(moment().format("DD MMM YYYY"));
+        $("#trip-start-time").html(moment().format("hh : mm A"));
+        var vehicle = _.find(OGPLVehicles, (o) => o.code == loadedVehicle) || {};
+        $(".last-odometer").html(vehicle.lastOdometer);
+      },
+  });
+  return true;
+}
+
+function hideStartTripDialogs() {
+  $("#start-trip-dialog").hide();
+}
